drivers/kw2xrf/kw2xrf_netdev.c

```diff
@@ -0,0 +1,765 @@
+        if ((len + ptr->iov_len + IEEE802154_FCS_LEN) > KW2XRF_MAX_PKT_LENGTH) {
+            DEBUG("[kw2xrf]: error: packet too large (%u byte) to be send\n",
+                  (unsigned)len + IEEE802154_FCS_LEN);
+            return -EOVERFLOW;
+        }
+        len = kw2xrf_tx_load(pkt_buf, ptr->iov_base, ptr->iov_len, len);
+    }
+
+    /* make sure ongoing t or tr sequenz are finished */
+    if (kw2xrf_can_switch_to_idle(dev)) {
+        kw2xrf_set_sequence(dev, XCVSEQ_IDLE);
+        dev->pending_tx++;
+    }
+    else {
+        /* do not wait, this can lead to a dead lock */
+        return 0;
+    }
+
+    /*
+     * Nbytes = FRAME_LEN - 2 -> FRAME_LEN = Nbytes + 2
+     * MKW2xD Reference Manual, P.192
+     */
+    dev->buf[0] = len + IEEE802154_FCS_LEN;
+
+    /* Help for decision to use T or TR sequenz */
+    _send_last_fcf = dev->buf[1];
+
+    kw2xrf_write_fifo(dev, dev->buf, dev->buf[0]);
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.tx_bytes += len;
+#endif
+
+    /* send data out directly if pre-loading id disabled */
+    if (!(dev->netdev.flags & KW2XRF_OPT_PRELOADING)) {
+        kw2xrf_tx_exec(dev);
+    }
+
+    return (int)len;
+}
+
+static int _recv(netdev2_t *netdev, void *buf, size_t len, void *info)
+{
+    kw2xrf_t *dev = (kw2xrf_t *)netdev;
+    size_t pkt_len = 0;
+
+    /* get size of the received packet */
+    pkt_len = kw2xrf_read_dreg(dev, MKW2XDM_RX_FRM_LEN);
+
+    /* just return length when buf == NULL */
+    if (buf == NULL) {
+        return pkt_len + 1;
+    }
+
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.rx_count++;
+    netdev->stats.rx_bytes += pkt_len;
+#endif
+
+    if (pkt_len > len) {
+        /* not enough space in buf */
+        return -ENOBUFS;
+    }
+
+    kw2xrf_read_fifo(dev, (uint8_t *)buf, pkt_len + 1);
+
+    if (info != NULL) {
+        netdev2_ieee802154_rx_info_t *radio_info = info;
+        radio_info->lqi = ((uint8_t*)buf)[pkt_len];
+        radio_info->rssi = (uint8_t)kw2xrf_get_rssi(radio_info->lqi);
+    }
+
+    /* skip FCS and LQI */
+    return pkt_len - 2;
+}
+
+static int _set_state(kw2xrf_t *dev, netopt_state_t state)
+{
+    switch (state) {
+        case NETOPT_STATE_SLEEP:
+            kw2xrf_set_power_mode(dev, KW2XRF_DOZE);
+            break;
+        case NETOPT_STATE_IDLE:
+            kw2xrf_set_power_mode(dev, KW2XRF_AUTODOZE);
+            kw2xrf_set_sequence(dev, dev->idle_state);
+            break;
+        case NETOPT_STATE_TX:
+            if (dev->netdev.flags & KW2XRF_OPT_PRELOADING) {
+                kw2xrf_tx_exec(dev);
+            }
+            break;
+        case NETOPT_STATE_RESET:
+            kw2xrf_reset_phy(dev);
+            break;
+	case NETOPT_STATE_OFF:
+            /* TODO: Replace with powerdown (set reset input low) */
+            kw2xrf_set_power_mode(dev, KW2XRF_HIBERNATE);
+        default:
+            return -ENOTSUP;
+    }
+    return sizeof(netopt_state_t);
+}
+
+netopt_state_t _get_state(kw2xrf_t *dev)
```

Comment:
should be static, otherwise name clash with other drivers, e.g. at86rf233, see Jenkins log [here](https://riot-ci.inet.haw-hamburg.de/job/RIOT/job/PR-5469/3/artifact/error_iotlab-a8-m3_linux_driver_tests.log)